packages/teleport/src/lib/tdp/playerClient.ts

```diff
@@ -34,6 +39,21 @@ export class PlayerClient extends Client {
     this.emit(PlayerClientEvent.TOGGLE_PLAY_PAUSE);
   }
 
+  // Overrides Client implementation.
+  processMessage(buffer: ArrayBuffer) {
+    const raw = this.textDecoder.decode(buffer);
+    if (raw === PlayerClientEvent.SESSION_END) {
+      this.emit(PlayerClientEvent.SESSION_END);
+      return;
+    }
+
+    const json = JSON.parse(raw);
```

Comment:
Could put a try/catch around this will throw if `raw` is incorrectly formatted.

Reply:
We could, although I'm not sure what we'd want to do if we caught it besides throw it.